src/Controls/src/Core/AppThemeBinding.cs

```diff
@@ -150,20 +141,45 @@ object GetValue()
 
 		void SetAttached(bool value)
 		{
-			var app = Application.Current;
-			if (app != null && _attached != value)
+			if (_attached == value)
+				return;
+
+			_attached = value;
+
+			if (_weakTarget?.TryGetTarget(out var target) == true && target is VisualElement ve)
 			{
+				// Use the VisualElement as this is faster than Application.RequestedThemeChanged
+				// by a significant margin: https://github.com/dotnet/maui/issues/18505
+				//
+				// We do "lose" a feature where all bindings would listen to the (at the time
+				// of first Apply) the current application. However, this has the drawback where
+				// the list of subscribers grows so large it takes too long to add/remove handlers.
+				//
+				// This logic here does also have a strong reference to the target object when
+				// applied, however this does not appear to be a problem in my tests. I also
+				// tested with making the _weakTarget field be a normal reference and still
+				// did not leak.
+
+				if (value)
+					ve.RequestedThemeChanged += OnRequestedThemeChanged;
+				else
+					ve.RequestedThemeChanged -= OnRequestedThemeChanged;
```

Comment:
This is _not_ a weak event, even though the target object is a weak reference. My tests pass, so I am not sure if there was a reason that is gone now and we can trust this test, or it was defensive coding. I checked the OG PR from forms and it was always a weak event: 

https://github.com/dotnet/maui/commit/529c8e90d0c22f3473e30889aa87e278b291a788#diff-a465dad5f7c3fe7c7ae4a52720c7ca8846cf243930297f34d15036a4397d1ea9R162-R171

and so was the OG implemntation of `AppThemeBinding`:

https://github.com/dotnet/maui/commit/6c40121e9798e9dd30bcaa7378fb0359d89ff868#diff-c0e940f176dec7f36005d68d0754e576b9ba344b21b6be8b4348028acd07f256R7

Reply:
`RequestedThemeChanged` uses `WeakEventManager`, right? That means `+=` won't keep strong references.

I think it should keep using `WeakEventManager` as random customer code might use that event.